### ISSUE TITLE
Proxy Authentication Required

### ISSUE THREAD

**Issue by tiwari-abhishek** [juunas11/aspnetcore2aadauth#8]
I am behind my company's proxy and firewall and I cannot get past the proxy. I have intentionally removed my tenant-id from the below message. I get this:

> An unhandled exception occurred while processing the request.
> HttpRequestException: Response status code does not indicate success: 407 (Proxy Authentication Required).
> 
> System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
> IOException: IDX10804: Unable to retrieve document from: 'https://login.microsoftonline.com/your-tenant-id/.well-known/openid-configuration'.
> 
> Microsoft.IdentityModel.Protocols.HttpDocumentRetriever+<GetDocumentAsync>d__8.MoveNext()
> InvalidOperationException: IDX10803: Unable to obtain configuration from: 'https://login.microsoftonline.com/your-tenant-id/.well-known/openid-configuration'.
> 
> Microsoft.IdentityModel.Protocols.ConfigurationManager+<GetConfigurationAsync>d__24.MoveNext()

I have tried both IIS Express and self host but nothing works.

In earlier .net versions it was possible to use application wide webproxy and I have working samples of Authenticating with Azure AD using .NET 4.6.2.

So what is it that I need to do to get through proxy? I was hoping the authentication to happen in the browser which would take care of authentication atleast, I would still however be stuck with token exchange as that needs a backchannel over a proxy.

Any thoughts?

**Comment by juunas11** [comment]
Yeah, the OpenID Connect handler needs to get the configuration for the endpoint through the backchannel.

This issue is probably a bit related to yours: https://github.com/dotnet/cli/issues/3065.

I'm no expert in this stuff, it might be a good question for Stack Overflow :)

**Comment by juunas11** [comment]
Oh cool to know you got it working. I was not aware of that either :) Might actually use that to setup a diagnostics listener for those.
